.travis.yml

```diff
@@ -0,0 +1,305 @@
+            - cmake-data
+            - g++-5
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+
+    - env: GCC_VERSION=5 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *gcc5
+
+    - env: GCC_VERSION=5 CXXFLAGS="-std=c++1z"
+      os: linux
+      addons: *gcc5
+
+    # gcc-6
+    - env: GCC_VERSION=6 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &gcc6
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - g++-6
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+
+    - env: GCC_VERSION=6 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *gcc6
+
+    - env: GCC_VERSION=6 CXXFLAGS="-std=c++1z"
+      os: linux
+      addons: *gcc6
+
+    # gcc-7
+    - env: GCC_VERSION=7 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &gcc7
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - g++-7
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+
+    - env: GCC_VERSION=7 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *gcc7
+
+    - env: GCC_VERSION=7 CXXFLAGS="-std=c++1z"
+      os: linux
+      addons: *gcc7
+
+    # clang-3.7
+    - env: CLANG_VERSION=3.7 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &clang37
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - clang-3.7
+            - libstdc++-5-dev
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.7
+
+    - env: CLANG_VERSION=3.7 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *clang37
+
+    # clang-3.8
+    - env: CLANG_VERSION=3.8 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &clang38
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - clang-3.8
+            - libstdc++-5-dev
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.8
+
+    - env: CLANG_VERSION=3.8 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *clang38
+
+    # clang-3.9
+    - env: CLANG_VERSION=3.9 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &clang39
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - clang-3.9
+            - libstdc++-6-dev
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.9
+
+    - env: CLANG_VERSION=3.9 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *clang39
+
+    - env: CLANG_VERSION=3.9 CXXFLAGS="-std=c++1z"
+      os: linux
+      addons: *clang39
+
+    # clang-4
+    - env: CLANG_VERSION=4.0 CXXFLAGS="-std=c++11"
+      os: linux
+      dist: trusty
```

Comment:
We should be on `trusty` by default now: https://blog.travis-ci.com/2017-08-31-trusty-as-default-status